keanu-project/src/test/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensorTest.java

```diff
@@ -383,4 +383,56 @@ public void canBroadcastDivide() {
         assertEquals(expected, diff);
     }
 
+    @Test
+    public void doesPositiveDivisionCorrectly() {
+        assertDropsFractionCorrectlyOnDivision(5, 3);
+    }
+
+    @Test
+    public void doesNegativeDivisionCorrectly() {
+        assertDropsFractionCorrectlyOnDivision(-5, 3);
+    }
+
+    @Test
+    public void canStartAsMaxInteger() {
```

Comment:
I think the tests could give us even more confidence if we didn't just concentrate on the max/min values in the integer range, but instead ran over a subset of the integer range (after all, the problem isn't that the single precision float didn't cover the integer range - it's more that it can't represent all of the numbers within that range).  I wrote a quick program and Double is able to represent every 32bit integer so we should be fine - but can't hurt to check a wider range?